src/execution_tree/primitives/argmin.cpp

```diff
@@ -0,0 +1,280 @@
+//  Copyright (c) 2018 Parsa Amini
+//  Copyright (c) 2018 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/execution_tree/primitives/argmin.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/util/matrix_iterators.hpp>
+
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/naming.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/throw_exception.hpp>
+#include <hpx/util/iterator_facade.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <limits>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <blaze/Math.h>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace phylanx { namespace execution_tree { namespace primitives
+{
+    ///////////////////////////////////////////////////////////////////////////
+    primitive create_argmin(hpx::id_type const& locality,
+        std::vector<primitive_argument_type>&& operands,
+        std::string const& name, std::string const& codename)
+    {
+        static std::string type("argmin");
+        return create_primitive_component(
+            locality, type, std::move(operands), name, codename);
+    }
+
+    match_pattern_type const argmin::match_data =
+    {
+        hpx::util::make_tuple("argmin",
+        std::vector<std::string>{"argmin(_1, _2)"},
+        &create_argmin, &create_primitive<argmin>)
+    };
+
+    ///////////////////////////////////////////////////////////////////////////
+    argmin::argmin(
+        std::vector<primitive_argument_type> && operands,
+        std::string const& name, std::string const& codename)
+        : primitive_component_base(std::move(operands), name, codename)
+    {}
+
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail
+    {
+        struct argmin : std::enable_shared_from_this<argmin>
+        {
+            argmin(std::string const& name, std::string const& codename)
+                : name_(name)
+                , codename_(codename)
+            {}
+
+        protected:
+            std::string name_;
+            std::string codename_;
+
+        protected:
+            using arg_type = ir::node_data<double>;
+            using args_type = std::vector<arg_type>;
+
+            primitive_argument_type argmin0d(args_type && args) const
+            {
+                return 0ul;
+            }
+
+            ///////////////////////////////////////////////////////////////////////////
+            primitive_argument_type argmin1d(args_type && args) const
+            {
```

Comment:
Don't we need to handle the axis here (same as argmax)?

Reply:
Done.